core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportClusterAllocationExplainAction.java

```diff
@@ -161,6 +166,9 @@ public static NodeExplanation calculateNodeExplanation(ShardRouting shard,
         if (node.getId().equals(assignedNodeId)) {
             finalDecision = ClusterAllocationExplanation.FinalDecision.ALREADY_ASSIGNED;
             finalExplanation = "the shard is already assigned to this node";
+        } else if (shard.unassigned() && hasPendingAsyncFetch) {
```

Comment:
We should take allocatedPostIndexCreate into account here (see PrimaryShardAllocator / ReplicaShardAllocator), i.e.
```
if (shard.unassigned() && shard.allocatedPostIndexCreate(indexMetaData) && hasPendingAsyncFetch) {
```

For replica shards (`ReplicaShardAllocator`) it's even a little bit more complex. Shard fetching is only relevant if not all deciders say YES (i.e. throttling leads to shard store data not being fetched) . We should thus include the condition `nodeDecision.type() != Decision.Type.YES` for replica shards to be exact.

```
if (shard.primary() == false && shard.unassigned() && shard.allocatedPostIndexCreate(indexMetaData) && nodeDecision.type() != Decision.Type.YES) {
    finalExplanation = "the shard cannot be assigned because allocation deciders say " + nodeDecision.type.name();
    finalDecision = ClusterAllocationExplanation.FinalDecision.NO
} else if (shard.unassigned() && shard.allocatedPostIndexCreate(indexMetaData) && hasPendingAsyncFetch) {
   ... shard fetching
}
```